### PR TITLE
 [Improvement][sparkLoad] use system env configs when user does not set env configs.

### LINKED ISSUE #21836: [Enhancement] (sparkLoad) spark load job can not get system env configs when user does not set env configs.
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

spark load job can not get system env configs when user does not set env configs. For example, report error： yarn application kill failed. Error: JAVA_HOME is not set and could not be found.

a more reasonable solution is using system env configs when user does not set env configs.

### Solution

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by CodeCooker17**
## Proposed changes

Issue Number: close #21836

use system env configs when user does not set env configs.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by CodeCooker17** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 58.13 seconds
 stream load tsv:          507 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         19 seconds loaded 2358488459 Bytes, about 118 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.2 seconds inserted 10000000 Rows, about 342K ops/s
 storage size: 17169425031 Bytes

**Comment by CodeCooker17** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 47.53 seconds
 stream load tsv:          535 seconds loaded 74807831229 Bytes, about 133 MB/s
 stream load json:         21 seconds loaded 2358488459 Bytes, about 107 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.1 seconds inserted 10000000 Rows, about 343K ops/s
 storage size: 17161876060 Bytes

**Comment by adonis0147** [comment]
run p0

File: fe/fe-core/src/main/java/org/apache/doris/catalog/SparkResource.java [hunk 1]
```diff
@@ -159,16 +160,17 @@ public Map<String, String> getSparkConfigs() {
     }
 
     public Map<String, String> getEnvConfigsWithoutPrefix() {
-        Map<String, String> envConfig = Maps.newHashMap();
+        Map<String, String> systemEnv = System.getenv();
+        Map<String, String> currentEnvConfigs = new HashMap<>(systemEnv);
```
Comment (adonis0147) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/SparkResource.java:164]: This line includes all system environment variables. What about including ones with `ENV_PREFIX` prefixes?

**Comment by CodeCooker17** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/catalog/SparkResource.java [hunk 2]
```diff
@@ -159,16 +161,30 @@ public Map<String, String> getSparkConfigs() {
...
+        Map<String, String> systemEnv = System.getenv();
+        Map<String, String> systemEnvWithPrefix = new HashMap<>();
+        for (Entry<String, String> entry : systemEnv.entrySet()) {
+            String key = entry.getKey();
+            String value = entry.getValue();
+            if (key.startsWith(ENV_PREFIX)) {
+                systemEnvWithPrefix.put(key, value);
+            }
+        }
+        return systemEnvWithPrefix;
```
Comment (adonis0147) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/SparkResource.java:187]:
```suggestion
    public Map<String, String> getEnvConfigsWithoutPrefix() {
        Map<String, String> envConfig = System.getenv().entrySet().stream()
                .filter(entry -> entry.getKey().startsWith(ENV_PREFIX))
                .collect(Collectors.toMap(Entry::getKey, Entry::getValue));
        if (envConfigs != null) {
            for (Map.Entry<String, String> entry : envConfigs.entrySet()) {
                if (entry.getKey().startsWith(ENV_PREFIX)) {
                    String key = entry.getKey().substring(ENV_PREFIX.length());
                    envConfig.put(key, entry.getValue());
                }
            }
        }
        return envConfig;
```

**Review by adonis0147 — CHANGES_REQUESTED** [review]

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 50.76 seconds
 stream load tsv:          547 seconds loaded 74807831229 Bytes, about 130 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.1 seconds inserted 10000000 Rows, about 343K ops/s
 storage size: 17161997569 Bytes

**Comment by CodeCooker17** [comment]
run buildall

**Comment by CodeCooker17** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 47.59 seconds
 stream load tsv:          544 seconds loaded 74807831229 Bytes, about 131 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          28.9 seconds inserted 10000000 Rows, about 346K ops/s
 storage size: 17161965668 Bytes

**Comment by adonis0147** [comment]
run p0

**Review by HonestManXin — APPROVED** [review]

**Review by adonis0147 — APPROVED** [review]
LGTM

**Pull request merged**